Underlight/cArts.cpp

```diff
@@ -5574,25 +5574,43 @@ void cArts::ApplyAbjure(int skill, lyra_id_t caster_id)
 					}
 					else
 					{
-						_stprintf(message, disp_message, timed_effects->name[i], player->Name());
+						if (n->IsPMare() == false)
```

Comment:
Please simplify this spaghetti code. 

This is the simplification:
const TCHAR* playerName = n->IsPMare() ? ( player->IsMale() ? "A male dreamer" : "A female dreamer" ) : player->Name();

Now you don't need 5 if-statements and the same code copied and pasted.

Reply:
If you don't like nesting the ternary operator then you can do this:
TCHAR* playerName = NULL;
if( n->IsPMare() )
    playerName = player->IsMale() ? "A male dreamer" : "A female dreamer";
else
    playerName = player->Name();

I prefer the first because it's a one-liner, but however you want. Then you have a single call to _stprintf:
_stprintf( message, disp_message, timed_effects->name[ i ], playerName );
gs->Talk( message, RMsg_Speech::SYSTEM_WHISPER, caster_id );

And that's it.